Add "node-map-max" to allow configuring nodemap size.

tommyp1ckles opened the pull request:
Using the node-map-max flag, one can now override the default 16k node map size.
This may be needed for large clusters, where the number of distinct node IPs in the cluster exceeds the standard size.

In helm, this will allow overriding the size using `bpf.nodeMapMax: <size>`

tommyp1ckles committed:
helm: add bpf.nodeMapMax helm val to configure node map size.

This can be used to override the default node-map-max value which sets bpf node map size.
In some cases, node map size may need to be overridden for very large clusters.

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

tommyp1ckles committed:
nodemap: add node-map-max flag to configure nodemap bpf size.

Using the node-map-max flag, one can now override the default 16k node map size.
This may be needed for large clusters, where the number of distinct node IPs in the cluster exceeds the standard size.

Also provides Size() to nodemap.Map interface such that loader can use this to set the NODE_MAP_MAX var while building bpf programs.

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

tommyp1ckles committed:
nodemap: add validation to check that node map max is at least 16384.

This is the constant default size prior to adding the flag.
There's not much reason to lower this value so to avoid edge cases we'll just say that this is the lower bound.

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

pchaigno commented on pkg/maps/nodemap/cell.go:
```diff
@@ -14,10 +15,24 @@ var Cell = cell.Module(
+}
+
+func (c Config) Flags(fs *pflag.FlagSet) {
+	fs.Uint32("node-map-max", defaultConfig.NodeMapMax,
```
It looks like all other sibling flags are prepended with `bpf-` (cf. https://docs.cilium.io/en/latest/network/ebpf/maps/#ebpf-maps). Do we want to do the same here?

pchaigno reviewed (commented):
I think we need documentation in https://docs.cilium.io/en/latest/network/ebpf/maps/#ebpf-maps. A few generated files are also missing but the CI already highlighted it.

pchaigno reviewed (changes requested):
These flags are documented in https://docs.cilium.io/en/latest/network/ebpf/maps/#ebpf-maps, which will need to be updated.

tommyp1ckles committed:
docs: add section for scale implications of nodemap size.

With previous commits adding the ability to adjust nodemap size, this adds a section explaining the implications of the nodemap sizing.

Signed-off-by: Tom Hadlaw <tom.hadlaw@isovalent.com>

qmonnet reviewed (approved):
Thanks!

qmonnet reviewed (commented):
The documentation workflow complains, please update the command reference:

```console
$ make -C Documentation update-cmdref
```
